src/Makefile.blocksds.arm7

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: CC0-1.0
+#
+# SPDX-FileContributor: Antonio Niño Díaz, 2023
+
+BLOCKSDS	?= /opt/blocksds/core
+BLOCKSDSEXT	?= /opt/blocksds/external
+
+# Source code paths
+# -----------------
+
+INCLUDEDIRS	:= . nds
+BINDIRS		:=
+
+# Defines passed to all files
+# ---------------------------
+
+DEFINES		:=
+
+# Libraries
+# ---------
+
+LIBS		:= -lnds7
+LIBDIRS		:= $(BLOCKSDS)/libs/libnds
+
+# Build artifacts
+# -----------------
+
+BUILDDIR	:= blocksds-build/arm7
+NAME		:= arm7
+ELF		:= blocksds-build/$(NAME).elf
+DUMP		:= blocksds-build/$(NAME).dump
+MAP		:= blocksds-build/$(NAME).map
+
+# Tools
+# -----
+
+PREFIX		:= arm-none-eabi-
+CC		:= $(PREFIX)gcc
+CXX		:= $(PREFIX)g++
+OBJDUMP		:= $(PREFIX)objdump
+MKDIR		:= mkdir
+RM		:= rm -rf
+
+# Verbose flag
+# ------------
+
+ifeq ($(VERBOSE),1)
+V		:=
+else
+V		:= @
+endif
+
+# Source files
+# ------------
+
+SOURCES_C	:= main-nds-arm7.c
+
+# Compiler and linker flags
+# -------------------------
+
+DEFINES		+= -DNDS -D__NDS__ -DARM7
+
+ARCH		:= -mcpu=arm7tdmi -mtune=arm7tdmi
+
+WARNFLAGS	:= -Wall
+
+ifeq ($(SOURCES_CPP),)
+    LD		:= $(CC)
+    LIBS	+= -lc
+else
+    LD		:= $(CXX)
+    LIBS	+= -lstdc++ -lc
+endif
+
+INCLUDEFLAGS	:= $(foreach path,$(INCLUDEDIRS),-I$(path)) \
+		   $(foreach path,$(LIBDIRS),-I$(path)/include)
+
+LIBDIRSFLAGS	:= $(foreach path,$(LIBDIRS),-L$(path)/lib)
+
+ASFLAGS		+= -x assembler-with-cpp $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) \
+		   -ffunction-sections -fdata-sections
+
+CFLAGS		+= -std=gnu11 $(WARNFLAGS) $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) -Os \
+		   -ffunction-sections -fdata-sections \
+		   -fomit-frame-pointer
+
+CXXFLAGS	+= -std=gnu++14 $(WARNFLAGS) $(DEFINES) $(ARCH) \
+		   -mthumb -mthumb-interwork $(INCLUDEFLAGS) -Os \
+		   -ffunction-sections -fdata-sections \
+		   -fno-exceptions -fno-rtti \
+		   -fomit-frame-pointer
```

Comment:
Do we want to put any of these behind an NDS_DEBUG check? The old Makefile excludes at least `-fomit-frame-pointer` when setting that.

`-DNDEBUG` also seems to be missing.

Reply:
Resolved, and updated the file/RAM size comparison to take this into account.